bin/k9-cdk.ts

```diff
@@ -53,6 +54,35 @@ const k9BucketPolicyProps: k9.s3.K9BucketPolicyProps = {
 
 k9.s3.makeBucketPolicy(stack, "S3Bucket", k9BucketPolicyProps);
 
+const autoDeleteBucket = new s3.Bucket(stack, 'AutoDeleteBucket', {
+    bucketName: 'k9-cdk-auto-delete-test',
+    removalPolicy: RemovalPolicy.DESTROY,
+    autoDeleteObjects: true,
+});
+
+console.log(`original autoDeleteBucket.policy: ${autoDeleteBucket.policy}`);
+const k9AutoDeleteBucketPolicyProps: k9.s3.K9BucketPolicyProps = {
+    bucket: autoDeleteBucket,
+    k9DesiredAccess: new Array<k9.k9policy.AccessSpec>(
+        {
+            accessCapability: k9.k9policy.AccessCapability.AdministerResource,
+            allowPrincipalArns: administerResourceArns,
+        },
+        {
+            accessCapability: k9.k9policy.AccessCapability.ReadConfig,
+            allowPrincipalArns: readConfigArns,
+        },
+        {
+            accessCapability: k9.k9policy.AccessCapability.WriteData,
+            allowPrincipalArns: writeDataArns,
+        }
+    )
+};
+
+k9.s3.makeBucketPolicy(stack, 'AutoDeleteBucket', k9AutoDeleteBucketPolicyProps);
```

Comment:
@brentryan how does this usage look to you?

What do you think `k9.s3.makeBucketPolicy` should be named?

I'm leaning towards `k9.s3.grantAccess`

Reply:
`k9.s3.grantAccess` follows the conventions that CDK is using. The only thing that might be slightly different is most of "grantXXX()" methods are for applying policy changes to IAM. When it's a resource policy, it's less consistent I've found. It's usually something like bucket.addToResourcePolicy() ...

I think my personal preference is for the "grantXXX" style.

Reply:
Agree - I'll improve function names and merge this PR today.